app/components/layout/Flashes/index.js

```diff
@@ -16,18 +20,77 @@ class Flashes extends React.Component {
   }
 
   componentDidMount() {
-    FlashesStore.on("flash", this.handleStoreChange);
+    FlashesStore.on('flash', this.handleStoreChange);
+    PusherStore.on('unavailable', this.handleConnectionError);
+    PusherStore.on('connected', this.handleConnectionSuccess);
   }
 
   componentWillUnmount() {
-    FlashesStore.off("flash", this.handleStoreChange);
+    FlashesStore.off('flash', this.handleStoreChange);
+    PusherStore.off('unavailable', this.handleConnectionError);
+    PusherStore.off('connected', this.handleConnectionSuccess);
+  }
+
+  handleStoreChange = (payload) => {
+    this.setState({ flashes: this.state.flashes.concat(payload) });
+  };
+
+  hasConnectionErrorFlash() {
+    return this.state.flashes.length > 0 && this.state.flashes[0].id === FLASH_CONN_ERROR_ID;
   }
 
+  // show a flash when there's a push connection issue
+  handleConnectionError = () => {
```

Comment:
Pusher only sends this event after the connection times out (30 seconds) plus it fails to connect once (10 seconds), so it can only happen once in a 40 second period.